internal/local/environment.go

```diff
@@ -0,0 +1,56 @@
+package local
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/10gen/realm-cli/internal/cli/user"
+)
+
+// Environment contains the name, full filepath, and api keys of a profile
+type Environment struct {
+	Name        string
+	Filepath    string
+	Credentials user.Credentials
+}
+
+// LoadProfileEnvs returns a list of each profile's environment containing name, filepath, and api keys
+func LoadProfileEnvs() ([]Environment, error) {
+	dir, dirErr := user.HomeDir()
+	if dirErr != nil {
+		return nil, fmt.Errorf("failed to get CLI profiles: %w", dirErr)
+	}
+
+	dirEntryList, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get CLI profiles: %w", dirErr)
+	}
+
+	environments := make([]Environment, 0, len(dirEntryList))
+	for _, v := range dirEntryList {
+		if !strings.Contains(v.Name(), user.ProfileType) {
+			continue
+		}
+
+		name := strings.TrimSuffix(v.Name(), filepath.Ext(v.Name()))
```

Comment:
if we go with the extension style check above (and keep the extension in its own `var`), we could make this:

```go
name := v.Name()
ext := filepath.Ext(name)
if ext == "."+user.ProfileType { continue }
name = name[0:len(name)-len(ext)]
```